src/adapter/src/coord.rs

```diff
@@ -3933,6 +3932,28 @@ pub fn serve(
             pg_timestamp_oracle_params.apply(config);
         }
 
+        // Register a callback so whenever the MAX_CONNECTIONS or SUPERUSER_RESERVED_CONNECTIONS
+        // system variables change, we update our connection limits.
+        let connection_limit_callback: Arc<dyn Fn(&SystemVars) + Send + Sync> =
+            Arc::new(move |system_vars: &SystemVars| {
+                let limit: u64 = system_vars.max_connections().cast_into();
+                let superuser_reserved: u64 =
+                    system_vars.superuser_reserved_connections().cast_into();
+
+                // If superuser_reserved > max_connections, prefer max_connections.
+                let superuser_reserved = std::cmp::min(limit, superuser_reserved);
```

Comment:
If `superuser_reserved > max_connections` is true, then that means only super users could connect to the database right? i.e., all connections are reserved for superusers and normal users will be locked out. That feels like a bad enough situation that we should add an error log here.

Reply:
Great call, added a soft assert here!